datacube_alchemist/_utils.py

```diff
@@ -44,11 +44,16 @@ def _write_stac(
 ):
     out_dataset = serialise.from_path(metadata_path)
     stac_path = Path(str(metadata_path).replace("odc-metadata.yaml", "stac-item.json"))
+    # Madness in deferred destination logic
+    uri_base = dataset_assembler.names.destination_folder(Path(task.settings.output.location))
+    uri_base = str(uri_base) + '/'
```

Comment:
This sort of manual treating URLs as strings worries me a little bit, it's probably fine, but I prefer to reach for the `urllib` functions.

... Not that we're consistent about that everywhere anyway.